src/core/thread/router_table.cpp

```diff
@@ -596,6 +596,24 @@ void RouterTable::UpdateRoutes(const Mle::RouteTlv &aRouteTlv, uint8_t aNeighbor
                 neighbor->SetLinkQualityOut(linkQuality);
                 SignalTableChanged();
             }
+
+            // If in the `aRouteTlv` we see the neighboring router is
+            // claiming that it has no link to us, by setting its
+            // `GetLinkQualityOut()` towards us as `kLinkQuality0`, and
+            // our two-way link quality to this router is at least
+            // `kLinkQuality2`, we schedule a unicast MLE Advertisement
+            // to be sent to this neighbor. This helps expedite
+            // recovery from any temporary router link quality
+            // mismatch. Otherwise, the neighboring router will
+            // continue to advertise that it has no link to us until
+            // our next trickle timer-triggered Advertisement
+            // transmission (which can be up to 32 seconds later).
+
+            if ((aRouteTlv.GetLinkQualityOut(index) == kLinkQuality0) &&
```

Comment:
one thought here is, in addition to checking outgoing link quality is kLinkQuality0 can we also check incoming link quality in route TLV is also > kLinkQuality2?

Reply:
Thanks @sarveshkumarv3.  The code is already doing this check.

It is part of `neighbor->GetTwoWayLinkQuality() >= kLinkQuality2` check. 
- `GetTwoWayLinkQuality()` gives the minimum of the tracked `LinkQualityIn` and `LinkQualityOut` of `neighbor` entry, and
- We set the `LinkQualityOut` on `neighbor` from what we get from the neighboring router as we process its `RouteTlv`. The block of code right above this is doing the following (reading the `LinkQualityIn` from `RouteTlv` and then set it as `LinkQualityOut` on `neighbor`):
```c++
            LinkQuality linkQuality = aRouteTlv.GetLinkQualityIn(index);

            if (neighbor->GetLinkQualityOut() != linkQuality)
            {
                neighbor->SetLinkQualityOut(linkQuality);
                SignalTableChanged();
            }
```